### PR TITLE
Update tensorflow-addons to ~0.11

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
Update `tensorflow-addons` to `~0.11` to get rid of the warning

```
UserWarning: Tensorflow Addons supports using Python ops for all Tensorflow versions above or equal to 2.2.0 and strictly below 2.3.0 (nightly versions are not supported).
 The versions of TensorFlow you are currently using is 2.3.0 and is not supported.
Some things might work, some things might not.
If you were to encounter a bug, do not file an issue.
If you want to make sure you're using a tested and supported configuration, either change the TensorFlow version or the TensorFlow Addons's version.
You can find the compatibility matrix in TensorFlow Addon's readme:
https://github.com/tensorflow/addons
```

Also set `tfa.options.TF_ADDONS_PY_OPS = True` to avoid https://github.com/tensorflow/addons#gpu-and-cpu-custom-ops-1

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -117,7 +117,7 @@ PyJWT = "~1.7"
 python-dateutil = "~2.8"
 tensorflow = "~2.3"
 tensorflow_hub = "~0.8"
-tensorflow-addons = "~0.10"
+tensorflow-addons = "~0.11"
```
Comment (wochinge) [inline pyproject.toml:120]: `0.10` still works, doesn't it? It's just about the warning, isn't it?
```suggestion
tensorflow-addons = ">= 0.10, <= 0.12"
```

**Review by wochinge — APPROVED** [review]
👍

File: pyproject.toml [hunk 2]
```diff
@@ -117,7 +117,7 @@ PyJWT = "~1.7"
 python-dateutil = "~2.8"
 tensorflow = "~2.3"
 tensorflow_hub = "~0.8"
-tensorflow-addons = "~0.11"
+tensorflow-addons = ">= 0.10, <= 0.12"
```
Comment (wochinge) [inline pyproject.toml:120]: I think you have to run `poetry update` again as the hash should change for `poetry.lock`

Comment (tabergma) [reply pyproject.toml:120]: already on it :)

**Comment by tabergma** [comment]
@wochinge can you help me resolve `Warning: The lock file is not up to date with the latest changes in pyproject.toml. You may be getting outdated dependencies. Run update to update them.`. Locally everything is fine. I also updated my poetry version locally, but nothing seem to be working. Can you please try to update the poetry file locally? Thanks.

**Comment by tabergma** [comment]
I think I found the issue on my side. Somehow poetry was not fully updated. Hopefully it is working now.

**Comment by wochinge** [comment]
🎉 Sorry, was a bit swamped and just saw the message now.

**Pull request merged**